enhancements/olm-feature-gate.md

```diff
@@ -0,0 +1,395 @@
+
+- As an author I want to clearly set the expecations in terms of feature maturity and don't want features that are not mature to be activated by default.
+
+#### Stories: Consuming
+
+- As a user I want to understand the product development direction and its impact on my use of it hands-on.
+
+- As a cluster operator I want to avoid features that could lower the stability or consistency of my production clusters.
+
+- As a cluster operator I don't want to rely on features that are not yet mature.
+
+- As a cluster operator I want to be able to quickly deactivate a feature when it has been identified that it lowers the stability or consistency of my clusters.
+
+### Implementation Details
+
+#### OLMConfig
+
+A new version v2alpha1 of the OLMConfig API will be created and supported along v1. This version will introduce a list named `gates` for activation and deactivation of features:
+
+~~~
+spec:
+  type: object
+    properties:
+      features:
+        description: Features contains the list of configurable OLM features.
+        type: object
+        properties:
+          disableCopiedCSVs:
+            description: DisableCopiedCSVs is used to disable OLM's "Copied CSV" feature for operators installed at the cluster scope, where a cluster scoped operator is one that has been installed in an OperatorGroup that targets all namespaces. When reenabled, OLM will recreate the "Copied CSVs" for each cluster scoped operator.
+            type: boolean
+          gates:
+            description: gates are used to activate/deactivate features.
+            type: array
+            items:
+              description: A gate is a name/value pair that contains whether a specific feature is enabled or not.
+              type: object
+              required:
+                - name
+                - value
+              properties:
+                name:
+                  description: name identifies a feature.
+                  type: string
+                value:
+                  description: value is a boolean that specifies whether the feature is activated or not.
+                  type: boolean
+~~~
+
+The internal structure will look like this:
+~~~
+// Features contains the list of configurable OLM features.
+type Features struct {
+	DisableCopiedCSVs *bool `json:"disableCopiedCSVs,omitempty"`
+  // +optional
+	Activations []FeatureActivation `json:"activations,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,2,rep,name=activations"`
+}
+
+type FeatureActivation struct { or type FeatureGate struct {
+  // +patchMergeKey=name
+  // +patchStrategy=merge
+  Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+  Value string `json:"value" protobuf:"bytes,2,opt,name=value"`
+}
+~~~
+
+#### Feature gate
+
+The feature gate implementation will leverage existing [Kubernetes libraries](https://pkg.go.dev/k8s.io/component-base/featuregate).
```

Comment:
Does this imply that all OLM sub-projects need to take a direct dependency on the k8s.io/component-base package?